dvc/dagascii.py

```diff
@@ -61,9 +63,10 @@ def __init__(self, cols, lines):
     def draw(self):
         """Draws ASCII canvas on the screen."""
         if sys.stdout.isatty():  # pragma: no cover
-            from asciimatics.screen import Screen
-
-            Screen.wrapper(self._do_draw)
```

Comment:
Neat! I didn't even know that pydoc has a built-in pager! Btw, does it handle wide text? Could you show how a wide DAG would look like? Also, please don't forget to remove the unused code 🙂

Reply:
The wide DAG looks like this

![Screenshot from 2019-11-27 19-04-48](https://user-images.githubusercontent.com/298994/69748498-ce9dff80-1148-11ea-92f9-6339ecfbfb3c.png)

So, it's a bit broken and it's caused by `less`, see
https://superuser.com/questions/272818/how-to-turn-off-word-wrap-in-less

If I run it with `-S`, like this
```
dvc pipeline show --ascii run.dvc |less -S
```
works ok, see

![Screenshot from 2019-11-27 19-07-53](https://user-images.githubusercontent.com/298994/69748701-3f451c00-1149-11ea-8254-17d133d85933.png)

So it's not perfect.

Reply:
on a related note, it still looks indecipherably ambiguous to me. Any reason we don't print a nested list (like `tree` does for directories in linux)?

Reply:
> on a related note, it still looks indecipherably ambiguous to me. Any reason we don't print a nested list (like `tree` does for directories in linux)?

good point, but it should be a different task?

Reply:
yes it was more a question for @efiop

Reply:
Also using `less` with `--clear-screen` fixes this weird behaviour

![weird-jump](https://user-images.githubusercontent.com/298994/69750678-9e0c9480-114d-11ea-93bc-5f809ac100c0.gif)

Reply:
@xliiv Oops, sorry, missed this old comment thread. Ok, so is there a way to pass those flags to the pager? I think I've seen `-S` used by default in git's pager.

Reply:
> Ok, so is there a way to pass those flags to the pager?

@efiop if you see anything simpler than the code below, I'll use it

```python
import contextlib
import os
from distutils.spawn import find_executable

.. cut

    def draw(self):
        """Draws ASCII canvas on the screen."""
        lines = map("".join, self.canvas)
        less_found = find_executable('less') is not None
        pager_set = os.getenv('MANPAGER') or os.getenv('PAGER')
        if less_found and not pager_set:
            with set_env(MANPAGER='', PAGER='less --chop-long-lines --clear-screen'):
                pydoc.pager(os.linesep.join(lines))
        else:
            # we can print info here, like "Install `less` and add it to PATH to enable pager"
            # alternatively, we can just print DAG without pydoc.pager
            # sys.stdout.write(text)
            pydoc.pager(os.linesep.join(lines))


# https://stackoverflow.com/a/34333710/740067
@contextlib.contextmanager
def set_env(**environ):
    """
    Temporarily set the process environment variables.

    >>> with set_env(PLUGINS_DIR=u'test/plugins'):
    ...   "PLUGINS_DIR" in os.environ
    True

    >>> "PLUGINS_DIR" in os.environ
    False

    :type environ: dict[str, unicode]
    :param environ: Environment variables to set
    """
    old_environ = dict(os.environ)
    os.environ.update(environ)
    try:
        yield
    finally:
        os.environ.clear()
        os.environ.update(old_environ)
```

Reply:
@xliiv That works, but we also need to add a flag for `more` then 🙂At which point, we will be duplicating the logic for pager detection and will only be using `if use_pager` from pydoc's getpager. So maybe we would be better off just copying pydoc's `getpager` here, adjusting flags for `less` and `more` and that is it?

Reply:
Btw, would be interesting to see how output from ttypager looks, that pydocs falls into. If it is unusable, then we could simply not use it in our implementation.

Reply:
So i've made it as here
https://github.com/iterative/dvc/pull/2815#issuecomment-559948931

And it works like this
`linux`

![linux](https://user-images.githubusercontent.com/298994/69914351-a6154e80-1443-11ea-9c73-bf4690b52509.gif)

`linux` - no less

![image](https://user-images.githubusercontent.com/298994/69914383-220f9680-1444-11ea-85da-de9a42683a85.png)



`windows cmd`

![windows-cmd](https://user-images.githubusercontent.com/298994/69914352-aada0280-1443-11ea-9361-fd08820f1f62.gif)


`windows powershell`

![windows-ps](https://user-images.githubusercontent.com/298994/69914357-ae6d8980-1443-11ea-984c-e4f0ccdd7ff1.gif)